text/0000-lock-without-registry.md

```diff
@@ -0,0 +1,35 @@
+- Start Date: 2017-05-09
+- RFC PR:
+- Yarn Issue:
+
+# Summary
+
+The lockfile yarn.lock should not include the base registry (`https://registry.npmjs.org`).
+
+# Motivation
+
+In yarn.lock, the `resolved` field includes registry such as `https://registry.npmjs.org`.
+In China, most developers will set it to `https://registry.npm.taobao.org` for speed; but it seems slow for travis-ci and circleci.
+
+# Detailed design
+
+Replace the `resolved` by a `hash` field.
```

Comment:
> URLs from yarn.lock are used for building dependency graph and downloads links, it is a cache of some sort.

Could these be reconstructed as needed? Would there by a performance penalty by doing so?

> Also it would be good for Chinese developers to be able to consume and produce yarn.lock files without modifying them.

Using `registry-rewrite` seems like a good approach to allowing developers to consume existing `yarn.lock` files. (specifically, markets such as China)

My concerns with that approach:
* I'm still not sure how that prevents leaking registry URLs.
* There are more developers than projects. Educating developers to use `registry-rewrite` may be more costly than educating project owners.